packages/gestalt/src/TypeaheadOption.js

```diff
@@ -0,0 +1,133 @@
+// @flow strict
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import Box from './Box.js';
+import Text from './Text.js';
+import TapArea from './TapArea.js';
+
+type TextColors =
+  | 'green'
+  | 'pine'
+  | 'olive'
+  | 'blue'
+  | 'navy'
+  | 'midnight'
+  | 'purple'
+  | 'orchid'
+  | 'eggplant'
+  | 'maroon'
+  | 'watermelon'
+  | 'orange'
+  | 'darkGray'
+  | 'gray'
+  | 'lightGray'
+  | 'red'
+  | 'white';
+
+type OptionObject = {
+  label: string,
+  value: string,
+};
+
+type OptionProps = {|
+  option: OptionObject,
+  selected?: OptionObject | null,
+  searchField: string,
+  hoverColor: string,
+  textColor: TextColors,
+  backgroundColor: string,
+  handleSelect: OptionObject => void,
+  // getOptionRef: (HTMLElement | null) => void,
+|};
+
+const Option = ({
+  option,
+  selected,
+  searchField,
+  handleSelect,
+  hoverColor,
+  textColor,
+  // getOptionRef,
+  backgroundColor,
+}: OptionProps) => {
+  // Determine if the option is the current selected item
+  const isSelectedItem = JSON.stringify(option) === JSON.stringify(selected);
+
+  // Highlight the current selected item
+  const [hover, setHover] = useState(isSelectedItem);
+
+  const handleOnTap = () => {
+    if (handleSelect) handleSelect(option);
+  };
+
+  return (
+    <Box
+      // ref={ref => {
+      //   // Only send ref of selected item
+      //   if (selected) getOptionRef(ref);
+      // }}
+      width="100%"
+      display="flex"
+      role="option"
+      aria-selected={isSelectedItem}
```

Comment:
@simkessy could we
* Set the `margin` and `padding` on this surrounding `Box`
* Also set the `rounding` on `TapArea`?
That should improve the focus outline on each option

Current behavior
![typeahead-focus-outline](https://user-images.githubusercontent.com/127199/85080893-38323980-b17f-11ea-847f-a48a3ecb940e.gif)